packages/core/src/envelope.ts

```diff
@@ -101,12 +109,30 @@ export function createEventEnvelope(
   // TODO: This is NOT part of the hack - DO NOT DELETE
   delete event.sdkProcessingMetadata;
 
-  const envelopeHeaders = {
+  let envelopeHeaders: EventEnvelopeHeaders = {
```

Comment:
nit: Instead of mutating `envelopeHeaders`, let's pull this logic into a function that returns `EventEnvelopeHeaders`.

Reply:
Also good idea, also done!